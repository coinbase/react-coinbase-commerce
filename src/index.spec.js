import React from 'react';
import { renderToString } from 'react-dom/server';
import CoinbaseCommerceButton from './index';

test('SSR properly renders clickable link without errors', () => {
  const string = renderToString(<CoinbaseCommerceButton checkoutId="aaaa"/>);
  expect(string).toBe('<div data-reactroot=""><a href="https://commerce.coinbase.com" rel="external" title="Pay with Bitcoin, Bitcoin Cash, Litecoin, or Ethereum"><button>Buy With Crypto</button></a></div>');
});

test('it renders and matches its snapshot', () => {
  const wrapper  = shallow(
    <CoinbaseCommerceButton checkoutId={'aaaa'}/>
  );
  expect(wrapper.children().first().find('IFrame').length).toEqual(0);
  expect(wrapper.children().first().find('Button').length).toEqual(1);
  expect(wrapper).toMatchSnapshot();
});

test('it opens up an iframe when its button is clicked', () => {
  const wrapper = shallow(
    <CoinbaseCommerceButton checkoutId={'aaaa'}/>
  );

  wrapper.find('Button').simulate('click');
  expect(wrapper.children().find('IFrame').length).toEqual(1);
  expect(wrapper).toMatchSnapshot();
});
