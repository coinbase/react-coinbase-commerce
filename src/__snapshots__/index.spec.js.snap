// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it opens up an iframe when its button is clicked 1`] = `
<div>
  <a
    href="https://commerce.coinbase.com"
    onClick={[Function]}
    rel="external"
    title="Pay with Bitcoin, Bitcoin Cash, Litecoin, or Ethereum"
  >
    <Button
      onClick={[Function]}
    />
  </a>
  <IFrame
    checkoutId="aaaa"
    onError={[Function]}
    onModalClose={[Function]}
  />
</div>
`;

exports[`it renders and matches its snapshot 1`] = `
<div>
  <a
    href="https://commerce.coinbase.com"
    onClick={[Function]}
    rel="external"
    title="Pay with Bitcoin, Bitcoin Cash, Litecoin, or Ethereum"
  >
    <Button
      onClick={[Function]}
    />
  </a>
</div>
`;
